### PR TITLE
doc/user: document configurable group and transactional IDs for Kafka

### PR DESCRIPTION
**Pull request by benesch**
This commit adds documentation for the features added in #23792. See that PR for details.

Do not merge until December 21.

### Motivation

   * This PR adds documentation.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

**Comment by guswynn** [comment]
@benesch added a commit documenting the new `START OFFSET` behavior

**Comment by benesch** [comment]
Thank you, @guswynn!

File: doc/user/content/sql/create-sink/kafka.md [hunk 1]
```diff
@@ -56,9 +56,12 @@ _item&lowbar;name_ | The name of the source, table or materialized view you want
...
 ### `CONNECTION` options
 
-Field                | Value  | Description
----------------------|--------|------------
-`TOPIC`              | `text` | The prefix used to generate the Kafka topic name to create and write to.
+Field                      | Value  | Description
+---------------------------|--------|------------
+`TOPIC`                    | `text` | The name of the Kafka topic to write to.
+`TRANSACTIONAL ID`         | `text` | Sets the transactional ID to use when producing to the Kafka topic.<br>Default: `materialize-{REGION ID}-{CONNECTION ID}-{SINK ID}`
+`PROGRESS GROUP ID PREFIX` | `text` | Sets the prefix of the consumer group ID to use when reading from the progress topic.<br>Default: `materialize-{REGION ID}-{CONNECTION ID}-{SINK ID}`
```
Comment (morsapaes) [inline doc/user/content/sql/create-sink/kafka.md:63]:
```suggestion
`TRANSACTIONAL ID`         | `text` | The transactional ID to use when producing to the Kafka topic.<br>Default: `materialize-{REGION ID}-{CONNECTION ID}-{SINK ID}`
`PROGRESS GROUP ID PREFIX` | `text` | The prefix of the consumer group ID to use when reading from the progress topic.<br>Default: `materialize-{REGION ID}-{CONNECTION ID}-{SINK ID}`
```

File: doc/user/content/sql/create-source/kafka.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ The same syntax, supported formats and features can be used to connect to a [Red
 Field                                | Value     | Description
 -------------------------------------|-----------|-------------------------------------
 `TOPIC`                              | `text`    | The Kafka topic you want to subscribe to.
+`GROUP ID PREFIX`                    | `text`    | Sets the prefix of the consumer group ID. See [Monitoring consumer lag](#monitoring-consumer-lag).<br>Default: `materialize-{REGION-ID}-{CONNECTION-ID}-{SOURCE_ID}`
```
Comment (morsapaes) [inline doc/user/content/sql/create-source/kafka.md:52]:
```suggestion
`GROUP ID PREFIX`                    | `text`    | The prefix of the consumer group ID. See [Monitoring consumer lag](#monitoring-consumer-lag).<br>Default: `materialize-{REGION-ID}-{CONNECTION-ID}-{SOURCE_ID}`
```

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 3]
```diff
@@ -285,7 +285,7 @@ The `mz_kafka_sources` table contains a row for each Kafka source in the system.
 | Field                  | Type           | Meaning                                                                                                   |
 |------------------------|----------------|-----------------------------------------------------------------------------------------------------------|
 | `id`                   | [`text`]       | The ID of the Kafka source. Corresponds to [`mz_catalog.mz_sources.id`](../mz_catalog#mz_sources).        |
-| `group_id_base`        | [`text`]       | The prefix of the group ID that Materialize will use when consuming data for the Kafka source.            |
+| `group_id_prefix`      | [`text`]       | The prefix of the group ID that Materialize will use when consuming data for the Kafka source.            |
```
Comment (morsapaes) [inline doc/user/content/sql/system-catalog/mz_internal.md:288]:
```suggestion
| `group_id_prefix`      | [`text`]       | The prefix of the group ID to use when consuming data for the Kafka source.            |
```

**Review by morsapaes — APPROVED** [review]

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 3 above)
Comment (morsapaes) [reply doc/user/content/sql/system-catalog/mz_internal.md]: It isn't very useful, but I hate inconsistency more!

**Pull request merged**